Modules/_zoneinfo.c

```diff
@@ -2205,7 +2205,15 @@ find_ttinfo(zoneinfo_state *state, PyZoneInfo_ZoneInfo *self, PyObject *dt)
     }
 
     unsigned char fold = PyDateTime_DATE_GET_FOLD(dt);
-    assert(fold < 2);
```

Comment:
Assertion here is unwanted, even though it's optimized out for some mysterious reason.

Reply:
Assertions are always optimized out :)